src/modules/facility-fixed-fee/facility-fixed-fee.controller.ts

```diff
@@ -23,7 +34,55 @@ export class FacilityFixedFeeController {
   @ApiInternalServerErrorResponse({
     description: 'An internal server error has occurred.',
   })
-  async getFixedFeesForFacility(@Param() params: GetFacilityFixedFeeParamsDto): Promise<GetFacilityFixedFeeResponse> {
+  async getFixedFeesForFacility(@Param() params: FacilityFixedFeeParamsDto): Promise<GetFacilityFixedFeeResponse> {
     return await this.facilityFixedFeeService.getFixedFeesForFacility(params.facilityIdentifier);
   }
+
+  @Post('facilities/:facilityIdentifier/fixed-fees')
+  @ApiOperation({
+    summary: 'Create a new fixed fee for a facility.',
+  })
+  @ApiBody({
+    type: CreateFacilityFixedFeeRequestItem,
+    isArray: true,
+  })
+  @ApiCreatedResponse({
+    description: 'The fixed fee has been successfully created.',
+    type: CreateFacilityFixedFeeResponse,
+  })
+  @ApiNotFoundResponse({
+    description: 'The facility was not found.',
+  })
+  @ApiBadRequestResponse({
+    description: 'Bad request.',
+  })
+  @ApiInternalServerErrorResponse({
+    description: 'An internal server error has occurred.',
+  })
+  async createFixedFeeForFacility(
+    @Param() params: FacilityFixedFeeParamsDto,
+    @Body(new ParseArrayPipe({ items: CreateFacilityFixedFeeRequestItem }))
+    newCreateFacilityFixedFeeRequest: CreateFacilityFixedFeeRequest,
+  ): Promise<CreateFacilityFixedFeeResponse> {
+    const facility = await this.facilityService.getFacilityByIdentifier(params.facilityIdentifier);
+
+    if (facility.facilityStageCode !== '07') {
```

Comment:
We do a check of whether or not `facilityStageCode` is issued / not issued elsewhere in our code, it'd be good if we can commonise them

Reply:
We are checking it in just 2 places, so I didn't create helper for it, maybe if it will be used more.

I changed created enum for facility stages.

Reply:
Okay thanks!